### PR TITLE
posix: test isDirEmpty, change error conditional

### PR DESCRIPTION
**Pull request by brendanashworth**
## Description
This commit adds a new test for `isDirEmpty` (for code coverage) and
changes around the error conditional. Previously, there was a `return
nil` statement that would only be triggered under a race condition and
would trip up our test coverage for no reason. With this new error
conditional, there's no awkward 'else'-esque condition, which means test
coverage will not change between runs for no reason in this specific
test. It's also a cleaner read.

## Motivation and Context
Fixes `cmd/posix.go` in https://github.com/minio/minio/issues/4658.

## How Has This Been Tested?
Tests have been added.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by aead — APPROVED** [review]
LGTM, nice find!

File: cmd/posix_test.go [hunk 1]
```diff
@@ -73,6 +73,31 @@ func TestPosixGetDiskInfo(t *testing.T) {
 	}
 }
 
+func TestPosixIsDirEmpty(t *testing.T) {
```
Comment (vadmeste) [inline cmd/posix_test.go:76]: Can you add a successful test case? an empty directory and isEmpty() returning true.

**Review by vadmeste — CHANGES_REQUESTED** [review]
One minor comment

File: cmd/posix_test.go (hunk 1 above)
Comment (brendanashworth) [reply cmd/posix_test.go:76]: Ah yes of course! Added, thanks.

**Review by vadmeste — APPROVED** [review]
LGTM
